tests/test_binary.py

```diff
@@ -0,0 +1,76 @@
+## -*- coding: utf-8 -*-
+#
+# Copyright (c) 2019, Veritomyx, Inc.
+#
+# This file is part of the Python SDK for PeakInvestigator
+# (http://veritomyx.com) and is distributed under the terms
+# of the BSD 3-Clause license.
+
+import unittest
+
+from context import peakinvestigator
+
+from peakinvestigator.io import save_binary
+
+import tempfile, os, six, filecmp
+
+class TestBinary(unittest.TestCase):
+
+    def setUp(self):
+        self.testempty = ['$x\x9cc`\x80\x00\x00\x00\x08\x00\x01\n']
+        self.testheader = [
+            '# This is a header\n# it should contain meta data\n# There is no limit to number of lines, and is optional.\n$x\x9cc`\x80\x00\x00\x00\x08\x00\x01\n']
+
+        self.testexample = [
+            '# This is a header\n# it should contain meta data\n# There is no limit to number of lines, and is optional.\n$x\x9cc`\x00\x03f\x87\xc9^`\x86C\xc0\x8f\xaf\x87\xfac4\x81|\xdb\x82\xc5\xd7\xb9\x1c\x82w\xce\x04\x82Y@\xbe\xcb\xec\xa5\x8fC\x1c\xc2\x1c\xc0\xea\x00\xbb\xdc\x11\xba\n']
+
+    def test_empty(self):
+        fd, filename = tempfile.mkstemp()
```

Comment:
Is there a reason you went with `mkstemp` here and below? `tempfile` also has `TemporaryFile` that is auto-deleted at the end, which seems like a cleaner solution (i.e. less code!).

Reply:
I think the Pythonic way to use `TemporaryFile` is something like:

```
with tempfile.TemporaryFile() as file:
  save_binary(...)
```